fix(overlay-panel): Fix overlay-panel escape key event handler.

TheRealJon opened the pull request:
## Description
Remove `keypress` event binding from overlay panel controller and add a check in `keydown` event handler to ensure preventDefault() has not already been called on the event. This ensures that if another handler has already handled the escape keydown event and explicitly prevented default behavior, then the overlay panel will not be dismissed.

Fixes #604

/kind bug

TheRealJon committed:
fix(overlay-panel): Fix overlay-panel escape event handler.

Overlay panel now only handles "keydown" and not "keypress". Also
checks that preventDefault() has not been called on escape keydown
events before handling them.

spadgett commented on src/components/overlay-panel/overlay-panel.controller.ts:
```diff
@@ -43,21 +43,26 @@ export class OverlayPanelController implements angular.IController {
+      // Only close this overlay if another handler has not called 
+      // preventDefault() on this event. This means only one overlay/modal will
+      // be closed per escape keydown event.
+      if(!event.isDefaultPrevented()) {
```
We should double check browser support here. Is this a jQuery event object? If so, it should be OK

https://api.jquery.com/event.isDefaultPrevented/#event-isDefaultPrevented

The standard is `event.defaultPrevented`, though.

https://developer.mozilla.org/en-US/docs/Web/API/Event/defaultPrevented

I'm assuming `$document.bind` will pass a jQuery event. `bind()` seems to be deprecated in jQuery 3.0 incidentally.

http://api.jquery.com/bind/

nit: space between `if` and `(`

spadgett reviewed (commented):
Thanks @TheRealJon, just one question

TheRealJon commented:
Regarding the `event.isDefaultPrevented()` it looks like jQuery just uses the javascript event's `defaultPrevented` property.

https://github.com/jquery/jquery/blob/3.2.1/src/event.js#L527

TheRealJon commented:
@spadgett I removed the deprecated jQuery `bind()` and `unbind()` functions and replaced them with the equivalent jQuery `on()` and `off()` functions. They do pass a jQuery event object to the handler, so using `event.isDefaultPrevented()` is probably the best bet. It actually looks like jQuery provides more cross-browser compatibility than the standard js flag.
